editor/src/messages/portfolio/document/utility_types/network_interface.rs

```diff
@@ -699,60 +774,51 @@ impl NodeNetworkInterface {
 		};
 
-		let mut output_types = Vec::new();
-
 		// If the node is not a protonode, get types by traversing across exports until a proto node is reached.
 		match &implementation {
 			graph_craft::document::DocumentNodeImplementation::Network(internal_network) => {
-				for export in internal_network.exports.iter() {
-					match export {
-						NodeInput::Node {
-							node_id: nested_node_id,
-							output_index,
-							..
-						} => {
-							let nested_output_types = self.output_types(nested_node_id, &[network_path, &[*node_id]].concat());
-							let Some(nested_nodes_output_types) = nested_output_types.get(*output_index) else {
-								log::error!("Could not get nested nodes output in output_types");
-								return Vec::new();
-							};
-							output_types.push(nested_nodes_output_types.clone());
-						}
-						NodeInput::Value { tagged_value, .. } => {
-							output_types.push(Some((tagged_value.ty(), TypeSource::TaggedValue)));
-						}
-
-						NodeInput::Network { .. } => {
-							// https://github.com/GraphiteEditor/Graphite/issues/1762
-							log::error!("Network input type cannot be connected to export");
-							return Vec::new();
-						}
-						NodeInput::Scope(_) => todo!(),
-						NodeInput::Inline(_) => todo!(),
-						NodeInput::Reflection(_) => todo!(),
+				let Some(export) = internal_network.exports.get(output_index) else {
+					return (concrete!(()), TypeSource::Error("Could not get export index"));
+				};
+				match export {
+					NodeInput::Node {
+						node_id: nested_node_id,
+						output_index,
+						..
+					} => self.output_type(nested_node_id, *output_index, &[network_path, &[*node_id]].concat()),
+					NodeInput::Value { tagged_value, .. } => (tagged_value.ty(), TypeSource::TaggedValue),
+					NodeInput::Network { .. } => {
+						// let mut encapsulating_path = network_path.to_vec();
+						// let encapsulating_node = encapsulating_path.pop().expect("No imports exist in document network");
+						// self.input_type(&InputConnector::node(encapsulating_node, *import_index), network_path)
+						(concrete!(()), TypeSource::Error("Could not type from network"))
 					}
+					NodeInput::Scope(_) => todo!(),
+					NodeInput::Inline(_) => todo!(),
+					NodeInput::Reflection(_) => todo!(),
 				}
 			}
 			graph_craft::document::DocumentNodeImplementation::ProtoNode(protonode) => {
 				let node_id_path = &[network_path, &[*node_id]].concat();
-				let primary_output_type = self.resolved_types.types.get(node_id_path).map(|ty| (ty.output.clone(), TypeSource::Compiled)).or_else(|| {
-					let node_types = random_protonode_implementation(protonode)?;
-					Some((node_types.return_value.clone(), TypeSource::RandomProtonodeImplementation))
-				});
-
-				output_types.push(primary_output_type);
-			}
-			graph_craft::document::DocumentNodeImplementation::Extract => {
-				output_types.push(Some((concrete!(()), TypeSource::Error("extract node"))));
+				self.resolved_types
```

Comment:
function no longer returns an option, will refactor in my next pr so that errors will display "Unknown" as the type